methcomp/mountain.py

```diff
@@ -0,0 +1,242 @@
+        * quantile - value along difference as function of quantile
+        * auc - area under curve
+        * median - quantile at median
+        * mountain_median - mountain at median
+        * iqr - interquantile range: index at low=50-iqr/2, high=50+iqr/2
+        * mountain_iqr - mountain low and high iqr
+
+    Examples
+    --------
+    Compare method 1 to method 2 and 3:
+
+    >>> import matplotlib.pyplot as plt
+    >>> from methcomp import mountain
+    >>> method1 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17,
+    >>>    18, 19, 20]
+    >>> method2 = [1.03, 2.05, 2.79, 3.67, 5.00, 5.82, 7.16, 7.69, 8.53, 10.38,
+    >>>    11.11, 12.17, 13.47, 13.83, 15.15, 16.12, 16.94, 18.09, 19.13, 19.54]
+    >>> method3 = [0.82, 2.09, 4.31, 4.18, 4.85, 6.01, 7.73, 7.95, 9.78, 10.07,
+    >>>    11.64, 11.88, 12.90, 13.55, 14.57, 16.82, 16.95, 17.62, 18.44, 20.14]
+    >>> mountain.Mountain(method1, method2, n_percentiles=500).plot(
+    >>>    color="blue", label="$M_1$ - $M_2$", unit="ng/ml")
+    >>> mountain.Mountain(method1, method3, n_percentiles=500).plot(
+    >>>    color="green", label="$M_1$ - $M_3$", unit="ng/ml")
+    >>> plt.show()
+
+    This will superimpose the comparisons by default, as plot creates it's own
+    axis if necessary, and reuses the current one if in place.
+
+    References
+    ----------
+    [krouwer_1995] Krouwer, Jan S., and Katherine L. Monti.
+                   "A simple, graphical method to evaluate laboratory assays."
+                   European journal of clinical chemistry and clinical biochemistry
+                   33.8 (1995): 525-528.
+    [monti_1995] Monti, Katherine L.
+                 "Folded empirical distribution function curves—mountain plots."
+                 The American Statistician 49.4 (1995): 342-345.
+    [xue_2011] Xue, Jing-Hao, and D. Michael Titterington.
+               "The p-folded cumulative distribution function and the mean absolute
+               deviation from the p-quantile." Statistics & probability letters
+               81.8 (2011): 1179-1182.
+    """
+
+    def __init__(
+        self,
+        method1: Union[List[float], np.ndarray],
+        method2: Union[List[float], np.ndarray],
+        n_percentiles: int = 100,
+        iqr: float = 68.27,
+    ):
+        """Construct a regressor.
+
+        Parameters
+        ----------
+        method1 : Union[List[float], np.ndarray]
+            Values for method 1
+        method2 : Union[List[float], np.ndarray]
+            Values for method 2
+        n_percentiles : int, optional
+            Number of percentile streps - more gives a smoother mountain plot
+            (default n=100)
+        iqr : float, optional
+            Interquartile range for to show in plot. If <0 iqr will be skipped.
+            (default: 90)
+        """
+        self.n_percentiles = n_percentiles
+        self.iqr = iqr
+        # Process args
+        super().__init__(method1, method2)
+
+    def plot(
```

Comment:
I don't want to sound whiny, but can we move the `plot` functionality to the end of the class? It seems more logical for me, because calculation is done "before" plotting

Reply:
Agree

Reply:
Also when doing PR it is the reviewers obligation to sound whiny :D